### DISCUSSION TITLE
Error notify plugin

### DISCUSSION THREAD

**Discussion by hnsantu554** [strongswan/strongswan#2456] [Q&A]
Error notify plugin specifies many error but does not provide information when it occurs 
Please help me to understand about below errors

ERROR_NOTIFY_PARSE_ERROR_HEADER = 4,
	ERROR_NOTIFY_PARSE_ERROR_BODY = 5,

thanks for the help

**Comment by tobiasbrunner** [comment]
As the names and corresponding error messages suggest, they occur when an IKE message's header or body can't be parsed correctly, respectively. The latter could have various reasons, [see here](https://github.com/strongswan/strongswan/blob/cf7fb47788dfb83bb5d8bd0bffdb582e381a2f0a/src/libcharon/sa/ikev2/task_manager_v2.c#L1639-L1680).

**Reply by hnsantu554** [reply]
Thanks for the info , Does strongswan leave a log regarding what kind of error occurred ex : icv failure or decryption error ?

**Reply by tobiasbrunner** [reply]
You can see the log messages if you follow the link in my post.

**Reply by hnsantu554** [reply]
> You can see the log messages if you follow the link in my post.

Thanks for the input ,I am using strongswan in linux , most of the operations are handled by kernel.
could you please confirm if icv and encryption decryption is handled by strongswan or linux kernel , thank you for your support

**Reply by tobiasbrunner** [reply]
IKE is handled by strongSwan, ESP by Linux. Only errors with IKE messages are reported by the _error-notify_ plugin.

**Reply by hnsantu554** [reply]
Can we use libipsec userland implementation provided by strongswan in linux?, does it provide specify error log event?

**Reply by tobiasbrunner** [reply]
I wouldn't recommend using it except for very special use cases. And while it does have log messages for error situations, it won't provide any events or counters. Consider using the kernel's audit framework instead if you need reporting of such error cases.

**Reply by hnsantu554** [reply]
We have a special case where our customer expecting specific error logs for AH/ESP, packet drop logs , i hope we can register for log events for each sub system using vici. Can we use libipsec instead of xfrm ? Does strongswan provides support for erro log events to client application if commercial license is taken?

**Reply by tobiasbrunner** [reply]
The commercial license does not include any additional code.

**Reply by hnsantu554** [reply]
> As the names and corresponding error messages suggest, they occur when an IKE message's header or body can't be parsed correctly, respectively. The latter could have various reasons, [see here](https://github.com/strongswan/strongswan/blob/cf7fb47788dfb83bb5d8bd0bffdb582e381a2f0a/src/libcharon/sa/ikev2/task_manager_v2.c#L1639-L1680).

Hi tobias,
Thanks for the input 
I have gone through the code for IKE parse error. I could figure out reasons for the IKE Body parse error, but when i checked for IKE PARSE_ERROR_HEADER , i could not figure out the reasons for Header error , it just says received invalid ike header .

It will be really helpful if you can help me the find reasons for IKE header errors.

Please help me to understand,
Thanks in advance

**Reply by tobiasbrunner** [reply]
If the IKE header is invalid for some reason (message too short, invalid version, exchange type unsuitable for the version etc.).

**Reply by hnsantu554** [reply]
> If the IKE header is invalid for some reason (message too short, invalid version, exchange type unsuitable for the version etc.).

In RFC 7296 error message types there is an error for invalid message id (INVALID_MESSAGE_ID) for IKE header. Does strongswan provides error notification for invalid message id? Please confirm

**Reply by tobiasbrunner** [reply]
No, strongSwan does not support `SET_WINDOW_SIZE` (the error is only used in that context).
